src/m2n/PointToPointCommunication.cpp

```diff
@@ -384,6 +386,42 @@ void PointToPointCommunication::acceptConnection(std::string const &acceptorName
   _isConnected = true;
 }
 
+void PointToPointCommunication::acceptPreConnection(std::string const &acceptorName,
+                                                 std::string const &requesterName)
+{
+  PRECICE_TRACE(acceptorName, requesterName);
+  PRECICE_ASSERT(not isConnected(), "Already connected!");
+  PRECICE_CHECK(utils::MasterSlave::isMaster() || utils::MasterSlave::isSlave(),
+        "You can only use a point-to-point communication between two participants which both use a master. "
+            << "Please use distribution-type gather-scatter instead.");
+  
+  std::vector<int> localConnectedRanks = _mesh->getConnectedRanks();
+
+  if (localConnectedRanks.empty()) {
+    _isConnected = true;
+    return;
+  }
+
+  // Accept point-to-point connections (as server) between the current acceptor
+  // process (in the current participant) with rank `utils::MasterSlave::_rank'
+  // and (multiple) requester processes (in the requester participant).
+  auto c = _communicationFactory->newCommunication();
+  
+  c->acceptConnectionAsServer(
+      acceptorName,
+      requesterName,
+      utils::MasterSlave::getRank(),
+      localConnectedRanks.size());
+
+  _connectionDataVector.reserve(localConnectedRanks.size());
+
+  for (int & connectedRank : localConnectedRanks) {
```

Comment:
```suggestion
  for (int connectedRank : localConnectedRanks) {
```
`int` can be copied without problem.